packages/core/upload/server/services/metrics.js

```diff
@@ -49,9 +47,10 @@ module.exports = ({ strapi }) => ({
     */
 
     const folderLevelsArray = (
-      await knex(folderTable)
+      await strapi.db
+        .getConnection(folderTable)
         .select(
-          knex.raw(
+          strapi.db.connection.raw(
```

Comment:
why not getConnection() here ?

Reply:
Because `getConnection()` automatically adds the schema (`.addSchema()`) if it exists and `.raw` and `.addSchema` are incompatible.

`.addSchema` is same as `select`, `update` etc., it starts a regular query.

So :
- `connection`always returns a knex object
- `getConnection()` returns a knex object or a knex queryBuilder object (if there is a schema)

Reply:
Ok for the sake of clarity I'm wondering if we may want to expose a db.raw() directly later on. TBD not urgent

Reply:
I like the idea. I probably can do it now